app/Http/Controllers/CommunityController.php

```diff
@@ -94,4 +95,61 @@ public function postLive(Request $request)
 
         return Redirect::back();
     }
+
+    public function getSlack()
+    {
+        $isEligible = false;
+        $accepted = false;
+        $isInviteAccepted = false;
+        $mail = config('osu.emails.account');
+
+        if (Auth::check()) {
+            $user = Auth::user();
+
+            $isEligible = $user->isSlackEligible();
+            $accepted = $user->isSlackAccepted();
+
+            if ($accepted) {
+                $slackUser = $user->slackUser()->first();
+
+                if ($slackUser->slack_id !== null) {
+                    $isInviteAccepted = true;
+                }
+            }
+        }
+
+        return view('community.slack', compact('isEligible', 'accepted', 'isInviteAccepted', 'mail'));
+    }
+
+    public function postSlackAgree()
+    {
+        $message = 'errors.unknown';
+
+        if (Auth::check()) {
+            $user = Auth::user();
+
+            if ($user->isSlackEligible()) {
+                $token = config('slack.token');
+                $contents = file_get_contents("https://osu-public.slack.com/api/users.admin.invite?email={$user->user_email}&token={$token}&set_active=true");
+
+                if ($contents) {
```

Comment:
just check and immediately return on failure instead of nesting so many if/elses